lib/active_admin/form_builder.rb

```diff
@@ -137,5 +139,15 @@ def js_for_has_many(assoc, form_block, template, new_record, class_string)
       }
     end
 
+    def allow_destroy?(object, allow_destroy_option)
+      case allow_destroy_option
+      when Symbol
+        object.public_send allow_destroy_option
+      when Proc
+        allow_destroy_option.call(object)
+      else
+        allow_destroy_option
+      end
+    end
```

Comment:
You should use this https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/view_helpers/method_or_proc_helper.rb here